### PR TITLE
Room based implementation of Local track database layer

### LINKED ISSUE #532: Local track database layer is based on SqlBrite which has been depreceated
## Description of issue
The local track database is based on SqlBrite. Since the square has deprecated and archived the SqlBrite repository, there will be no more support.

## What is expected
Newly created local Track database layer based on modern architecture component. We will shift our database to ensure support and to be flexible for the modern components. We will use a SQlite database with Room ORM.

### PR DESCRIPTION
**Pull request by arvindnegi1**
# Description
Newly created track database layer based on android Room.

Fixes #532  

## Task
- [x] Local track database layer implementation based on Sqlite database with Room.
- [x] First test on real device.
- [x] Unit testing Room DAO classes.

### CODE REVIEW

File: org.envirocar.app/tests/android/java/org/envirocar/app/test/ObfuscationTest.java [hunk 1]
```diff
@@ -18,157 +18,157 @@
...
-
-    private Measurement createMeasurementNear(Measurement first, long time) {
-        Measurement result = new MeasurementImpl(first.getLatitude()+0.0001, first.getLongitude()+0.0001);
-        result.setTime(time);
-        return result;
-    }
-
-}
\ No newline at end of file
+//import android.test.InstrumentationTestCase;
```
Comment (dewall) [inline org.envirocar.app/tests/android/java/org/envirocar/app/test/ObfuscationTest.java:21]: please either remove classes that are commted out or integrate them properly

File: org.envirocar.app/tests/android/java/org/envirocar/app/test/ResourceLoadingTestCase.java [hunk 2]
```diff
@@ -16,32 +16,32 @@
...
-		sc.close();
-		return sb.toString();
-	}
-	
-	protected String readJsonAsset(String assetRes) throws IOException {
-		return readJson(getClass().getResourceAsStream(assetRes));
-	}
-	
-}
+///package org.envirocar.app.test;
```
Comment (dewall) [inline org.envirocar.app/tests/android/java/org/envirocar/app/test/ResourceLoadingTestCase.java:19]: same here

File: org.envirocar.app/tests/android/java/org/envirocar/app/test/TermsOfUseTest.java [hunk 3]
```diff
@@ -30,16 +30,16 @@
...
-	public void testTermsOfUseInstanceParsing() throws JSONException, IOException {
-		JsonElement gson = new Gson().fromJson(readJsonAsset("/terms_of_use_instance_mockup.json"), JsonElement.class);
-		TermsOfUse tou = new TermsOfUseSerde().deserialize(gson, null, null);
-		
-		Assert.assertTrue("Unexpected issuedDate", tou.getIssuedDate().equals("2022-06-09"));
-		Assert.assertTrue("Unexpected contents", tou.getContents().equals("v50..."));
-	}
-	
-}
+//public class TermsOfUseTest extends ResourceLoadingTestCase {
```
Comment (dewall) [inline org.envirocar.app/tests/android/java/org/envirocar/app/test/TermsOfUseTest.java:33]: same here

File: org.envirocar.app/tests/android/java/org/envirocar/app/test/UserTest.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
  */
 package org.envirocar.app.test;
 
-public class UserTest extends ResourceLoadingTestCase {
+//public class UserTest extends ResourceLoadingTestCase {
```
Comment (dewall) [inline org.envirocar.app/tests/android/java/org/envirocar/app/test/UserTest.java:21]: same here

**Comment by dewall** [comment]
@arvindnegi1 please fix

File: org.envirocar.app/tests/android/java/org/envirocar/app/test/TermsOfUseTest.java (hunk 3 above)
Comment (arvindnegi1) [reply org.envirocar.app/tests/android/java/org/envirocar/app/test/TermsOfUseTest.java]: sir, I revert these changes.

**Comment by arvindnegi1** [comment]
@bpross-52n sir please review.

**Comment by bpross-52n** [comment]
Thanks @arvindnegi1, I will review this. It could take a few days, though.

**Comment by bpross-52n** [comment]
Looks good, I will merge this today.

**Pull request merged**